modules/gdscript/gdscript_parser.h

```diff
@@ -577,20 +606,10 @@ class GDScriptParser {
 					case ENUM:
 						return m_enum->get_datatype();
 					case ENUM_VALUE: {
-						// Always integer.
-						DataType out_type;
-						out_type.type_source = DataType::ANNOTATED_EXPLICIT;
-						out_type.kind = DataType::BUILTIN;
-						out_type.builtin_type = Variant::INT;
-						return out_type;
+						return enum_value.identifier->get_datatype();
```

Comment:
This change looks super sensible, and makes `ENUM_VALUE` and `SIGNAL` conform to all the other types.

I guess the concern is just that there might be a situation in which the identifiers haven't been fully resolved, at which point this would return garbage, right? Just a reminder to double-check that this will never be null.

Reply:
well, `member.get_datatype()` should (in this pr) only be called after `resolve_class_member()` was called, otherwise it will return garbage or a resolving type (which is intentional), 
datatype resolution should always be done before this method is called
ill put that in the description :)
also enum values don't get added if parsing the id fails